src/Components/CommandPrompt/CommandPrompt.tsx

```diff
@@ -0,0 +1,152 @@
+                return lowerCommand.includes(lowerTyped);
+            }),
+        );
+        setSelectedCommandIndex(matchedCommands.length >= 0 ? 0 : undefined);
+    }, [typed]);
+
+    useEffect(() => {
+        if (shown) {
+            inputField.current.focus();
+            setSelectedCommandIndex(0);
+        } else {
+            inputField.current.blur();
+            setTyped("");
+            if (commandToRun.current) {
+                commandToRun.current.callback();
+                commandToRun.current = null;
+            }
+        }
+    }, [shown]);
+
+    useEffect(() => {
+        document.addEventListener("keydown", handleKeyDownInDocument);
+
+        return () =>
+            document.removeEventListener("keydown", handleKeyDownInDocument);
+    }, [shown, selectedCommandIndex]);
+
+    const handleKeyDownInDocument = (event) => {
+        // TODO: Shortcut in a settings pane ?
+        if ((event.ctrlKey || event.metaKey) && event.key == "k") {
+            setShown(() => !shown);
+            event.preventDefault();
+        }
+
+        if (shown && event.key == "Escape") {
+            setShown(false);
+            event.preventDefault();
+        }
+
+        // Selected item is cyclic
+        switch (event.key) {
+            case "ArrowUp":
+                setSelectedCommandIndex(
+                    (selectedCommandIndex + 1) % matchedCommands.length,
+                );
+                event.preventDefault();
+                break;
+            case "ArrowDown":
+                setSelectedCommandIndex(
+                    (selectedCommandIndex - 1 + matchedCommands.length) %
+                        matchedCommands.length,
+                );
+                event.preventDefault();
+                break;
+        }
+    };
+
+    const handleValueChangeInInput = (event) => setTyped(event.target.value);
+
+    const handleInputKeyDown = (event) => {
+        if (event.key == "Enter") {
+            runSelectedCommand(
+                matchedCommands.length != 0 ? selectedCommandIndex : undefined,
+            );
+        }
+    };
+
+    const runSelectedCommand = (index?: number) => {
+        if (index === undefined) return;
+
+        commandToRun.current = matchedCommands[index];
+        setShown(false);
+    };
+
+    const renderMatchedCommands = () => {
+        let index = -1;
+        return matchedCommands.map((command) => {
+            index++;
+            const commandIndex = index;
+
+            const clickCallback = () => {
+                setSelectedCommandIndex(commandIndex);
+                runSelectedCommand(commandIndex);
+            };
+
+            return (
+                <CommandItem
+                    onClick={clickCallback}
+                    command={command}
+                    typed={typed}
+                    selected={selectedCommandIndex == commandIndex}
+                />
+            );
+        });
+    };
```

Comment:
I think that `.map((command) => …)` and `let index` can be simplified to `.map((command, i) => …)`